pkg/controller/daemon/update.go

```diff
@@ -112,7 +112,7 @@ func (dsc *DaemonSetsController) getUnavailableNumbers(ds *extensions.DaemonSet,
 		}
 		available := false
 		for _, pod := range daemonPods {
-			if v1.IsPodAvailable(pod, ds.Spec.MinReadySeconds, metav1.Now()) {
+			if v1.IsPodAvailable(pod, ds.Spec.MinReadySeconds, metav1.Now()) && pod.DeletionTimestamp == nil {
```

Comment:
IsPodAvailable is modeled after IsPodReady so if we made this change it should live either in both or in IsPodReady. We would have to vet such a change though because IsPodReady has to be used in a bunch of different places.